scripts/apps/search/directives/SaveSearch.ts

```diff
@@ -39,6 +39,12 @@ export function SaveSearch($location, asset, api, session, notify, $rootScope) {
                 scope.edit = null;
                 scope.activateSearchPane = false;
             };
+            scope.isValid = function(edit) {
+                if (edit.filter.query.raw == null) {
+                    return !!edit.name;
```

Comment:
It's easier to understand when you are more explicit. I assume the reason you are trying to convert a string to bool is to check if the string is empty. `typeof edit.name === 'string' && edit.name.length > 0` does exactly that and doesn't hide your intention :) The same applies to a line below.